cocotb/handle.py

```diff
@@ -855,7 +880,7 @@ def _set_value(self, value, call_sim):
                 "Unsupported type for integer value assignment: {} ({!r})"
                 .format(type(value), value))
 
```

Comment:
The validation needs to happen here before `call_sim` is called, otherwise the user won't get an error immediately.

Reply:
> The validation needs to happen here before `call_sim` is called, otherwise the user won't get an error immediately.

Are you suggesting that doing the range checks in C++ as you recommended earlier is not desired after all?

Reply:
Would you mind linking to where I said that? Apologies in advance if my opinions are inconsistent over time!

Reply:
Fully understand that opinions can change over time. Here's some links:
https://github.com/cocotb/cocotb/pull/1742#discussion_r419086506
https://github.com/cocotb/cocotb/pull/1742#discussion_r419086122